### PR TITLE
DZ use HashSet in GoCounting

### PR DESCRIPTION
**Pull request by dzegel**
Currently the tests are very brittle assuming elements are added to the lists in a particular order.
Using `HashSet` instead of `IEnumerable` and `Array` makes the tests more robust.

### CODE REVIEW

File: generators/Exercises/Generators/GoCounting.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ private string RenderTerritoryAssert()
         private string RenderTerritoriesAssert()
         {
             var territoriesAssert = new StringBuilder();
-            territoriesAssert.AppendLine(Render.AssertEqual("expected.Keys", "actual.Keys"));
+            territoriesAssert.AppendLine(Render.AssertEqual("expected.Keys.Count", "actual.Keys.Count"));
```
Comment (dzegel) [inline generators/Exercises/Generators/GoCounting.cs:75]: being that al the keys are used in the assertions below checking the count is sufficient.
alternative approaches would be to convert the key collections to sets or to just remove this assertion

Comment (ErikSchierboom) [reply generators/Exercises/Generators/GoCounting.cs:75]: I think: just remove the assertion.

**Comment by ErikSchierboom** [comment]
> still need to update the SUT's method signatures to return ISets instead of IEnumerables but I'm still trying to figure out how those files are generated

You mean the stub `GoCounting.cs` files? If so, they are not generated.

**Comment by ErikSchierboom** [comment]
@dzegel https://github.com/exercism/csharp/blob/main/generators/Exercises/Generators/GoCounting.cs is the generator class. You should be able to run it using `cd generators && dotnet run -e go-counting`

**Comment by dzegel** [comment]
> @dzegel https://github.com/exercism/csharp/blob/main/generators/Exercises/Generators/GoCounting.cs is the generator class. You should be able to run it using `cd generators && dotnet run -e go-counting`

I did run it, the test file changes in the diff are from running that

File: exercises/practice/go-counting/GoCounting.cs [hunk 2]
```diff
@@ -15,12 +15,12 @@ public GoCounting(string input)
...
     }
 
-    public Tuple<Owner, IEnumerable<(int, int)>> Territory((int, int) coord)
+    public Tuple<Owner, HashSet<(int, int)>> Territory((int, int) coord)
     {
         throw new NotImplementedException("You need to implement this function.");
     }
 
-    public Dictionary<Owner, IEnumerable<(int, int)>> Territories()
+    public Dictionary<Owner, HashSet<(int, int)>> Territories()
```
Comment (dzegel) [inline exercises/practice/go-counting/GoCounting.cs:23]: manually updated

File: exercises/practice/go-counting/GoCountingTests.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ public void Black_corner_territory_on_5x5_board()
             "  W  ";
         var sut = new GoCounting(board);
         var actual = sut.Territory(coordinate);
-        var expected = (Owner.Black, new[] { (0, 0), (0, 1), (1, 0) });
+        var expected = (Owner.Black, new HashSet<(int, int)> { (0, 0), (0, 1), (1, 0) });
```
Comment (dzegel) [inline exercises/practice/go-counting/GoCountingTests.cs:19]: all changes in this file were auto generated

File: generators/Exercises/Generators/GoCounting.cs (hunk 1 above)
Comment (dzegel) [reply generators/Exercises/Generators/GoCounting.cs]: done

**Review by ErikSchierboom — APPROVED** [review]
Perfect!

**Pull request merged**